gpii/node_modules/gpii-oauth2/gpii-oauth2-authz-server/src/AuthorizationService.js

```diff
@@ -137,3 +169,33 @@ gpii.oauth2.authorizationService.setSelectedPreferences = function (dataStore, u
     }
     // TODO else communicate not found?
 };
+
+gpii.oauth2.authorizationService.grantClientCredentialsAccessToken = function (dataStore, clientId, scope) {
+    // Record the credential client access token if we haven't already
+    var client = dataStore.findClientById(clientId);
+    if (!scope || scope.indexOf("add_preferences") === -1 || !client.allowAddPrefs) {
+        return false;
+    }
+
+    var clientCredentialsToken = dataStore.findClientCredentialsTokenByClientId(clientId);
+    if (!clientCredentialsToken) {
+        var accessToken = gpii.oauth2.authorizationService.generateAccessToken();
+        clientCredentialsToken = dataStore.addClientCredentialsToken({
+            clientId: clientId,
+            accessToken: accessToken,
+            allowAddPrefs: true
+        });
+    }
+
+    return clientCredentialsToken.accessToken;
+};
+
+gpii.oauth2.authorizationService.savePrefs = function (preferencesDataSource, preferences, view, callback) {
+    view = view || "";
+    var promise = preferencesDataSource.set({"view": view}, preferences);
+    promise.then(function (data) {
+        callback(data);
+    }, function (err) {
+        callback({"error": "Error when saving preferences"});
```

Comment:
Our standard form for such errors holds 
{
    isError: true,
    message: "Error when saving preferences"
}
but it is courteous to make some attempt to pass along information from the upstream error.
A common idiom I've used in my upcoming big branch is the following:
var error = $.extend(err, {
   message: "Error when saving preferences: " + err.message
});
callback(error)
although it seems odd to have the same callback for both errors and success.
This should really just be a function returning a promise